### ISSUE TITLE
Custom mods are not loading!

### ISSUE THREAD

**Issue by conceptfac** [Arkensor/DayZCommunityOfflineMode#385]
Custom mods are not loading! I creating a simple mod with notifacation "Hello World" but not load too BBP, CF, Dabs none are loading too

**Comment by Arkensor** [comment]
Not loading how? Did you add the to the batch file? Please provide enough information so you can be helped.

**Comment by griha41** [comment]
I don't know how and why, but I have no issues with mods on my offline mission. And offline mission is the best way for me to check is my mods working as expected.

I'm connecting everything what do i need, Expansion, TraderPlus (if need to create trader zone) and never met the issues woth offline mode. I think you shall be more careful about what are you doing and thene everything went fine.

**Comment by MercuryZxDEV** [comment]
Can confirm this issue, but only when using the .bat file.

I tried everything from 
- different Drives
- different folder setups

Funny thing, if I use the Vanilla Launcher with local Mod, it works.

**Comment by griha41** [comment]
> Can confirm this issue, but only when using the .bat file.
> 
> I tried everything from 
> - different Drives
> - different folder setups
> 
> Funny thing, if I use the Vanilla Launcher with local Mod, it works.

Folks then how the heck do I loading the mods!?

Maybe you had done something wrong?

-------
Let's try another way, show me up how do you including the mods. Predefine some folders:
DayZ exe folder = %DZ%

**Comment by MercuryZxDEV** [comment]
Some of the things I tried:
1. 
-mod="@LocalMod"
G:\SteamLibrary\steamapps\common\DayZ\ @LocalMod\addons\pbosLocatedHere.pbo


2.
-mod="P:\BloodragePacked"
P:\BloodragePacked\addons\pbosLocatedHere.pbo

-mod="C:\BloodragePacked"
C:\BloodragePacked\addons\pbosLocatedHere.pbo

I even tried it without the "addons" folder.

@griha41  I dont have a clue why its not working.

**Comment by griha41** [comment]
> Some of the things I tried:
> 1. 
> -mod="@LocalMod"
> G:\SteamLibrary\steamapps\common\DayZ\@LocalMod\addons\pbosLocatedHere.pbo
> 
> 2.
> -mod="P:\BloodragePacked"
> P:\BloodragePacked\addons\pbosLocatedHere.pbo
> 
> -mod="C:\BloodragePacked"
> C:\BloodragePacked\addons\pbosLocatedHere.pbo
> 
> I even tried it without the "addons" folder.
> 
> @griha41  I dont have a clue why its not working.
> 

First of all.
Relative patch P:\ is system-wide, but don't heck your mind with the crp.

Now please make that way as I written bellow:
1. Openup .bat script for DZ offline mission 
2. At the line start DayZ_x64.exe, right after the .exe append next text "-mod=@LocalMod" "-servermod=@LocalMod"

Why the hell shall we use " symbol - it's for the path and parameter protection.
Why the hell -servermod command line?
- We using it because I have no idea what is that mod, but in SP mission for DayZ the best option is including your mod in -servermod param.

Now about folder structure:
1. Locate your @LocalMod folder inside DZ folder.
2. @LocalMod create single folder "Addons".
3. Inside Addons folder shall be only .pbo and nothing more even bisign and other BI crp.

That's all about.
If you wish you can predefine like
"-mod=P:\Bi ToTal Idiots\@LocalMod"
In current case it doesn't matter, coz DZ will eat it quite simply.

Start the DayZ and then try to find item from that mod (or do what it supposed to do).

After you've done with DZ head up to:
Open Windows Explorer and input next address:
%Localappdata%\DayZ\

You'll find the DayZ Logs directory, open latest .RPT log and check Line 3, it SHALL contains your mod directory, if it's didn't then you've done something wrong with you bat file or started something else.

I checked that method about 3 hrs ago and it's still working...

**Comment by MercuryZxDEV** [comment]
Jesus @griha41  it literally was the mistake of

-mod="LocalMod"
"-mod=LocalMod"

Maybe this should be included in the landingpage "How to add Mods"
I literally watched a youtube video where it has been done wrong lmao.

Thanks a lot!

**Comment by griha41** [comment]
> Jesus @griha41  it literally was the mistake of
> 
> -mod="LocalMod"
> "-mod=LocalMod"
> 
> Maybe this should be included in the landingpage "How to add Mods"
> I literally watched a youtube video where it has been done wrong lmao.
> 
> Thanks a lot!

Yes man, I met that issue in Arma 3, in Arma 2 OA mods had been included by -mod="@Mod1;@Mod...;@ModX"

In Arma 3tye rule was changed to: "-mod=@Mod1;@Mod...;@ModX"

In DayZ before 1.04 you can start the game with -mod="@Mod1;@Mod...;@ModX", ainse 1.04 only with the "-mod=@Mod1;@Mod...;@ModX"

So always welcome!

**Comment by 1435thunder** [comment]
I couldnt get Deerisle to load - kept getting "No world called deerisle" or other mod script errors, because it cant locate the Mods.

So I used the DayZ launcher.
Then I checked the DayZ launcher logs to determine how the launcher composed and executed the Mods/Mission.

Example Launncher Log:
2024-11-17 10:44:57,114 [ INFO]   
1: GameExecutor: parameters:   -name=bubbles -filePatching -doLogs -profiles=F;\SteamLibrary\steamapps\common\DayZ\profiles -mission=F;\SteamLibrary\steamapps\common\DayZ\missions\DayZCommunityOfflineMode.deerisle "-mod=F:\SteamLibrary\steamapps\common\DayZ\!Workshop\@CF;F:\SteamLibrary\steamapps\common\DayZ\!Workshop\@DeerIsle"

I copied the Launchers format.

@echo off
taskkill /F /IM DayZ_x64.exe /T
RD /s /q "storage_-1" > nul 2>&1
cd /d "F:\SteamLibrary\steamapps\common\DayZ"

start DayZ_x64.exe "-mod=F;\SteamLibrary\steamapps\common\DayZ\Workshop\@CF;F\SteamLibrary\steamapps\common\DayZ\!Workshop\@DeerIsle" "-mission=F:\SteamLibrary\steamapps\common\DayZ\missions\DayZCommunityOfflineMode.deerisle" -filePatching -doLogs
 

To clarify 
"-mod=.\!Workshop\@blah blah;@blah blah;@blah blah"  < Dose not function for every Mod/Mission.

Declaring absolute paths for each mod resolves the script errors e.g.  
"-mod=F;\SteamLibrary\steamapps\common\DayZ\Workshop\@CF; F\SteamLibrary\steamapps\common\DayZ\!Workshop\@DeerIsle;"

Again if this is helpful the Launcher log is located 
C:\Users\youraccount\AppData\Local\DayZ Launcher\Logs\Launcher.log

**Comment by griha41** [comment]
Are you kidding? Will you bet that's everyone can remember his/her account name?
Blah use the short links:
%localappdata%\DayZ <- logs patch of DayZ game, can be overrode by "-profiles=SomeProfileName" parameter.

Don't think that's everyone has a had on a shoulders, peoples are mostly idiots, they don't give a crp about start-up parameters - they only copy -> paste and don't asking themselves why and what do they copied.
